docs/source/development/profiling_and_benchmarking.md

```diff
@@ -0,0 +1,272 @@
+(`profiling-database.csv`) and the `run_benchmarking.py` tool, which can be used to
+benchmark new profiling data. The basic process is:
+
+* read the current `prof/combined.prof` file and convert it to human-readable CSV data,
+* find the best performance for each call over recent previous runs, and
+* check if the relative performance of the incoming code is notably slower than that
+  best performance.
+
+By default, we use the 5 most recent code versions and the threshold performance is 5%
+slower than the previous best performance. The report tool consists of a command line
+wrapper around a set of profiling functions, which can be imported for programatic use.
+for use.
+
+The usage of the tool is:
+
+```text
+usage: run_benchmarking.py [-h] [--exclude EXCLUDE] [--n-runs N_RUNS]
+                           [--tolerance TOLERANCE] [--append-on-pass] [--new-database]
+                           [--plot-path PLOT_PATH]
+                           prof_path database_path fail_data_path label
+
+Run the package benchmarking.
+
+This function runs the standard benchmarking for the pyrealm package. The profiling
+tests in the test suite generate a set of combined profile data across the package
+functionality. This command then reads in a set of combined profile data and
+compares it to previous benchmark data.
+
+The profiling excludes all profiled code objects matching regex patterns provided
+using the `--exclude` argument. The defaults exclude standard and site packages,
+built in code and various other standard code, and are intended to reduce the
+benchmarking to only code objects within the package.
+
+positional arguments:
+  prof_path              Path to pytest-profiling output
+  database_path          Path to benchmarking database
+  fail_data_path         Output path for data on benchmark fails
+  label                  A text label for the incoming profiling results, typically a
+                         commit SHA
+
+options:
+  -h, --help             show this help message and exit
+  --exclude EXCLUDE      Exclude profiled code matching a regex pattern, can be repeated
+                         (default: ['{.*}', '<.*>', '/lib/'])
+  --n-runs N_RUNS        Number of most recent runs to use in benchmarking (default: 5)
+  --tolerance TOLERANCE  Tolerance of time cost increase in benchmarking (default: 0.05)
+  --append-on-pass       Add incoming data to the database when benchmarking passes
+                         (default: False)
+  --new-database         Use the incoming data to start a new profiling database
+                         (default: False)
+  --plot-path PLOT_PATH  Generate a benchmarking plot to this path (default: None)
+```
+
+### Manual benchmarking
+
+Once you have run the `pytest-profiling` test suite and generated `prof/combined.prof`
+for some new code, you can run the following code to benchmark those results. In the
+code below, `incoming` is used as a label for the new code - you could also a commit SHA
```

Comment:
```suggestion
code below, `incoming` is used as a label for the new code - you could also commit SHA
```

Reply:
I'm not sure if SHA was mentioned somewhere earlier of if it is a standard abbreviation, if not could you please explain?